File: gluecodium/src/test/resources/smoke/defaults/output/android/com/example/smoke/DefaultValues.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import com.example.NativeBase;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
public final class DefaultValues extends NativeBase {
    public static final class StructWithDefaults {
        public int intField;
        public long uintField;
        public float floatField;
        public double doubleField;
        public boolean boolField;
        @NonNull
        public String stringField;
        public StructWithDefaults() {
            this.intField = 42;
            this.uintField = 4294967295L;
            this.floatField = 3.14f;
            this.doubleField = -1.4142;
            this.boolField = true;
            this.stringField = "\\Jonny \"Magic\" Smith\n";
        }
    }
    public static final class NullableStructWithDefaults {
        @Nullable
        public Integer intField;
        @Nullable
        public Long uintField;
        @Nullable
        public Float floatField;
        @Nullable
        public Boolean boolField;
        @Nullable
        public String stringField;
        public NullableStructWithDefaults() {
            this.intField = null;
            this.uintField = null;
            this.floatField = null;
            this.boolField = null;
            this.stringField = null;
        }
    }
    public static final class StructWithSpecialDefaults {
        public float floatNanField;
        public float floatInfinityField;
        public float floatNegativeInfinityField;
        public double doubleNanField;
        public double doubleInfinityField;
        public double doubleNegativeInfinityField;
        public StructWithSpecialDefaults() {
            this.floatNanField = Float.NaN;
            this.floatInfinityField = Float.POSITIVE_INFINITY;
            this.floatNegativeInfinityField = Float.NEGATIVE_INFINITY;
            this.doubleNanField = Double.NaN;
            this.doubleInfinityField = Double.POSITIVE_INFINITY;
            this.doubleNegativeInfinityField = Double.NEGATIVE_INFINITY;
        }
    }
    public static final class StructWithEmptyDefaults {
        @NonNull
        public List<Integer> intsField;
        @NonNull
        public List<Float> floatsField;
        @NonNull
        public Map<Long, String> mapField;
        @NonNull
        public DefaultValues.StructWithDefaults structField;
        @NonNull
        public Set<String> setTypeField;
        public StructWithEmptyDefaults() {
            this.intsField = new ArrayList<>();
            this.floatsField = new ArrayList<>();
            this.mapField = new HashMap<>();
            this.structField = new DefaultValues.StructWithDefaults();
            this.setTypeField = new HashSet<>();
        }
    }
    public static final class StructWithTypedefDefaults {
        public long longField;
        public boolean boolField;
        @NonNull
        public String stringField;
        public StructWithTypedefDefaults() {
            this.longField = 42L;
            this.boolField = true;
            this.stringField = "\\Jonny \"Magic\" Smith\n";
        }
    }
    /**
     * For internal use only.
     * @hidden
     * @param nativeHandle The SDK nativeHandle instance.
     * @param dummy The SDK dummy instance.
     */
    protected DefaultValues(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    @NonNull
    public static native DefaultValues.StructWithDefaults processStructWithDefaults(@NonNull final DefaultValues.StructWithDefaults input);
}

File: gluecodium/src/test/resources/smoke/defaults/output/android-kotlin/com/example/smoke/DefaultValues.kt
/*

 *
 */

@file:JvmName("DefaultValuesExtensions")


package com.example.smoke

import com.example.NativeBase

class DefaultValues : NativeBase {

    class StructWithDefaults {
        @JvmField var intField: Int
        @JvmField var uintField: Long
        @JvmField var floatField: Float
        @JvmField var doubleField: Double
        @JvmField var boolField: Boolean
        @JvmField var stringField: String



        constructor() {
            this.intField = 42
            this.uintField = 4294967295L
            this.floatField = 3.14f
            this.doubleField = -1.4142
            this.boolField = true
            this.stringField = "\\Jonny \"Magic\" Smith\n"
        }





    }

    class NullableStructWithDefaults {
        @JvmField var intField: Int?
        @JvmField var uintField: Long?
        @JvmField var floatField: Float?
        @JvmField var boolField: Boolean?
        @JvmField var stringField: String?



        constructor() {
            this.intField = null
            this.uintField = null
            this.floatField = null
            this.boolField = null
            this.stringField = null
        }





    }

    class StructWithSpecialDefaults {
        @JvmField var floatNanField: Float
        @JvmField var floatInfinityField: Float
        @JvmField var floatNegativeInfinityField: Float
        @JvmField var doubleNanField: Double
        @JvmField var doubleInfinityField: Double
        @JvmField var doubleNegativeInfinityField: Double



        constructor() {
            this.floatNanField = Float.NaN
            this.floatInfinityField = Float.POSITIVE_INFINITY
            this.floatNegativeInfinityField = Float.NEGATIVE_INFINITY
            this.doubleNanField = Double.NaN
            this.doubleInfinityField = Double.POSITIVE_INFINITY
            this.doubleNegativeInfinityField = Double.NEGATIVE_INFINITY
        }





    }

    class StructWithEmptyDefaults {
        @JvmField var intsField: List<Int>
        @JvmField var floatsField: List<Float>
        @JvmField var mapField: Map<Long, String>
        @JvmField var structField: DefaultValues.StructWithDefaults
        @JvmField var setTypeField: Set<String>



        constructor() {
            this.intsField = mutableListOf()
            this.floatsField = mutableListOf()
            this.mapField = mutableMapOf()
            this.structField = DefaultValues.StructWithDefaults()
            this.setTypeField = mutableSetOf()
        }





    }

    class StructWithTypedefDefaults {
        @JvmField var longField: Long
        @JvmField var boolField: Boolean
        @JvmField var stringField: String



        constructor() {
            this.longField = 42L
            this.boolField = true
            this.stringField = "\\Jonny \"Magic\" Smith\n"
        }





    }



    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}







    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)


        @JvmStatic external fun processStructWithDefaults(input: DefaultValues.StructWithDefaults) : DefaultValues.StructWithDefaults
    }
}


